### DISCUSSION TITLE
why Notifications dont have async

### DISCUSSION THREAD

**Discussion by vmachacek** [umbraco/Umbraco-CMS#10588] [Features and ideas]
I wan to call API every-time I save media to Umbraco - tinifyer so that the image get compressed with some wicked algorithms I don't care about. 

This is not really possible in a nice way, because Notifications have synchronous API. 

Am I missing something?

**Comment by nul800sebastiaan** [comment]
Hi there @vmachacek - This sounds like an excellent question for the forums where our friendly community can help you find the best solution for your requirements. 
Make sure to head on over to https://our.umbraco.com and ask follow up questions there! 👍

**Reply by Shazwazza** [reply]
@vmachacek If you are referring to event notifications in v9, they do support async. They support both. See https://github.com/umbraco/Umbraco-CMS/blob/v9/dev/src/Umbraco.Core/Events/INotificationHandler.cs#L28
